src/neural_network/neural_network.py

```diff
@@ -47,11 +50,20 @@ def train(self, training_data, mini_batch_size, epochs_count, eta, momentum = 0,
                 self._update_mini_batch(mini_batch, eta, momentum)
             if test_data:
                 print("Epoch {0}: {1} / {2}".format(epoch, self.evaluate(test_data), n_test))
+                self._calculate_test_set_cost(test_data)
             else:
                 print("Epoch {0} complete".format(epoch))
+            for layer in self.layers:
+                layer.add_weights_to_history()
+                    
+            # save average cost
+            avg_batch_cost = self.summed_batch_cost / mini_batch_size
+            self.epoch_train_costs.append(avg_batch_cost)
+            
 
     def _update_mini_batch(self, mini_batch, eta, momentum):
         mini_batch_size = len(mini_batch)
+        self.summed_batch_cost = 0
```

Comment:
Czy to nie jest błąd? `summed_batch_cost` to suma kosztów dla wszystkich `mini_batch`. Zerowanie tego w każdym wywołaniu `_update_mini_batch` sprawi, że `summed_batch_cost` będzie kosztem tylko z ostatniego mini batcha

Reply:
Poprawione